httpx/concurrency/asyncio/backend.py

```diff
@@ -126,10 +122,12 @@ def is_connection_dropped(self) -> bool:
         # (For a solution that uses private asyncio APIs, see:
         # https://github.com/encode/httpx/pull/143#issuecomment-515202982)
 
-        return self.stream_reader.at_eof()
+        return self.stream.at_eof()
 
     async def close(self) -> None:
-        self.stream_writer.close()
+        # FIXME: We should await on this call, but need a workaround for this first:
+        # https://github.com/aio-libs/aiohttp/issues/3535
```

Comment:
With the `await`, the `test_start_tls_on_socket_stream` fails transiently with the error mentioned in this thread in about 1 out of 5-30 test runs (thanks, `pytest-repeat`) for me on Python 3.7.2, 3.7.4, and 3.8.0b4.

There is a possible fix (a custom exception handler set on the event loop) in the thread, but it's fairly complicated and I'm not sure how to integrate it nicely. This is no _worse_ than we had so I think it should be fixed in a separate PR.

Reply:
Wow, that thread doesn't look good. :fearful:

Reply:
Is it OK to not await this `.close()` call, though?

Reply:
Thanks, all these details are very helpful. :)

Since the 3.8 docs it’s only *possible* to await on .close() (which I find kind of weird, having in mind all those « coroutine was never awaited » exceptions that usually causes), I’m okay with keeping it as it is currently. As you said, we actually never waited for the socket to close before (and if we did, we’d have probably only encountered this issue earlier?).

Reply:
One (not very good) reason to include this `try`/`except` is that without it I don't think this change has 100% test coverage since `wait_closed` is never called.